docs/analyzer/checker_and_analyzer_configuration.md

```diff
@@ -191,3 +191,74 @@ configuration arguments (json in this case) should be in one line :
 ```
 -config="{ "Checks": "clang-diagnostic-*,clang-analyzer-*", "WarningsAsErrors": "", "HeaderFilterRegex": "", "AnalyzeTemporaryDtors": false, "CheckOptions": [ { "key": "google-readability-braces-around-statements.ShortStatementLines", "value": "1" }, { "key": "modernize-loop-convert.MaxCopySize", "value": "16" }, { "key": "modernize-loop-convert.NamingStyle", "value": "CamelCase" }, { "key": "modernize-use-nullptr.NullMacros", "value": "NULL" } ] }"
 ```
+
+# Cppcheck
+
+As of CodeChecker 6.20, Codechecker can now execute the Cppcheck analyzer.
+
+## Configuration
+
+The Cppcheck analyzer can be configured with --analyzer-config cppcheck:* paramterers.
+
+The supporte analyzer configuration items can be listed with `CodeChecker analyzers --analyzer-config cppcheck --details`
+
+As of the CodeChecker 6.20, the following options are supported:
+
+* `cppcheck:addons` A list of cppcheck addon files.
+* `cppcheck:libraries` A list of cppcheck library definiton files.
+* `cppcheck:platform` The platform configuration .xml file.
+
+Please note that for addons and libraries, multiple items can be specified in the following format: --analyzer-config cppcheck:addons <addon.py> --analyzer-config cppcheck:addons <addon2.py>.
+
+Cppcheck only supports a limited number of platforms. Custom bit lengths can be specified with a platform file.
+
+An example platform file from the cppcheck manual:
+
+``` xml
+<?xml version="1"?>
+<platform>
+  <char_bit>8</char_bit>
+  <default-sign>signed</default-sign>
+  <sizeof>
+    <short>2</short>
+    <int>4</int>
+    <long>4</long>
+    <long-long>8</long-long>
+    <float>4</float>
+    <double>8</double>
+    <long-double>12</long-double>
+    <pointer>4</pointer>
+    <size_t>4</size_t>
+    <wchar_t>2</wchar_t>
+  </sizeof>
+</platform>
+```
```

Comment:
Are the "common" platforms like LP64 (Unix) and LLP64 (Windows) modelled by Cppcheck by default? Or should we always generate this file if the user does not specify one? In the latter case, the generator code for this feature seems to be missing.

Reply:
The native platform is assumed if the config file is not provided.